packages/vite/src/plugins/composable-keys.ts

```diff
@@ -59,9 +59,11 @@ export const composableKeysPlugin = createUnplugin((options: ComposableKeysOptio
               break
           }
 
+          const endsWithComma = code.slice(codeIndex + (node as any).start, codeIndex + (node as any).end - 1).trim().endsWith(',')
```

Comment:
Perf: this is a heavy operation to run on each code block, copying the whole string, trimming (that also copies) to only check the last character is `,`. We could replace it with a look-behind loop.
```suggestion
          // TODO: Optimize me
          const endsWithComma = code.slice(codeIndex + (node as any).start, codeIndex + (node as any).end - 1).trim().endsWith(',')
```